File: src/store/options.ts
import { Ref, ref, watch } from "vue";
import { Store } from "@tauri-apps/plugin-store";

type Options = {
  quality: Quality;
  chunkSize: ChunkSize;
  automode: boolean;
};

enum Quality {
  HIGH = "high",
  MEDIUM = "medium",
  LOW = "low",
}

enum ChunkSize {
  LARGE = "large",
  MEDIUM = "medium",
  SMALL = "small",
}

const options: Ref<Options> = ref({
  quality: Quality.HIGH,
  chunkSize: ChunkSize.MEDIUM,
  automode: false,
});

const store = Store.load("settings.json");

// Initial loading
store.then((s) => s.get<Options>("options")).then((opt) => (options.value = opt || options.value));

watch(options, () => store.then((s) => s.set("options", options.value)), { deep: true });

export { Quality, ChunkSize, options };

File: src-tauri/src/options.rs
use crate::image_processing::{ChunkSize, Quality};
use serde::Deserialize;
use tauri_plugin_store::StoreExt;

#[derive(Deserialize)]
pub struct Options {
    pub quality: Quality,
    #[serde(rename(deserialize = "chunkSize"))]
    pub chunk_size: ChunkSize,
    pub automode: bool,
}

impl Default for Options {
    fn default() -> Self {
        Options {
            quality: Quality::High,
            chunk_size: ChunkSize::Medium,
            automode: false,
        }
    }
}

pub fn get_options(app: &tauri::AppHandle) -> Options {
    let store = match app.store("settings.json") {
        Ok(store) => store,
        Err(_) => return Default::default(),
    };

    store
        .get("options")
        .and_then(|options| serde_json::from_value::<Options>(options).ok())
        .unwrap_or_default()
}
